algoliasearch/account_client.go

```diff
@@ -0,0 +1,123 @@
+package algoliasearch
+
+import "fmt"
+
+type accountClient struct{}
+
+func NewAccountClient() AccountClient {
+	return &accountClient{}
+
+}
+
+func (a *accountClient) CopyIndex(src, dst Index) (err error) {
+	if src.GetAppID() == dst.GetAppID() {
+		return SameAppIDErr
+	}
+
+	if _, err := dst.GetSettings(); err == nil {
+		return IndexAlreadyExistsErr
+	}
+
+	// Copy objects
+	{
+		it, err := src.BrowseAll(nil)
+		if err != nil {
+			return fmt.Errorf("cannot browse source index records: %v", err)
+		}
+
+		var records []Object
+
+		for {
+			res, err := it.Next()
+			if err != nil {
+				if err == NoMoreHitsErr {
+					break
+				} else {
+					return fmt.Errorf("error while iterating source index records: %v", err)
+				}
+			}
+			records = append(records, Object(res))
+		}
+		err = dst.ReplaceAllObjects(records)
+		if err != nil {
+			return fmt.Errorf("error while replacing destination index records: %v", err)
+		}
+	}
+
+	var taskIDs []int
+
+	// Copy synonyms
+	{
+		it := NewSynonymIterator(src)
+
+		var synonyms []Synonym
+
+		for {
+			synonym, err := it.Next()
+			if err != nil {
+				if err == NoMoreSynonymsErr {
+					break
+				} else {
+					return fmt.Errorf("error while iterating source index synonyms: %v", err)
+				}
+			}
+			synonyms = append(synonyms, *synonym)
+		}
+
+		if synonyms != nil {
+			res, err := dst.ReplaceAllSynonyms(synonyms)
+			if err != nil {
+				return fmt.Errorf("error while replacing destination index synonyms: %v", err)
+			}
+			taskIDs = append(taskIDs, res.TaskID)
+		}
+	}
+
+	// Copy rules
+	{
+		it := NewRuleIterator(src)
+
+		var rules []Rule
+
+		for {
+			rule, err := it.Next()
+			if err != nil {
+				if err == NoMoreRulesErr {
+					break
+				} else {
+					return fmt.Errorf("error while iterating source index rules: %v", err)
+				}
+			}
+			rules = append(rules, *rule)
+		}
+		if rules != nil {
+			res, err := dst.ReplaceAllRules(rules)
+			if err != nil {
+				return fmt.Errorf("error while replacing destination index rules: %v", err)
+			}
+			taskIDs = append(taskIDs, res.TaskID)
+		}
+	}
+
+	settings, err := src.GetSettings()
+	if err != nil {
+		return fmt.Errorf("cannot retrieve source index settings: %v", err)
+	}
+
+	{
+		res, err := dst.SetSettings(settings.ToMap())
+		if err != nil {
+			return fmt.Errorf("cannot set destination index settings: %v", err)
+		}
+		taskIDs = append(taskIDs, res.TaskID)
+	}
+
+	for _, taskID := range taskIDs {
```

Comment:
The developer here don't have the possibility to choosing if he wants to wait or not. Can't you return `taskIDs` variable and the developer can wait afterwords if he wants?

Reply:
You're right. As discussed, fixing it by returning the list of taskIDs to let the user decide whether or not to wait on them. A `waitable`-like object will rather come in the next major release to make the client easier to use.